src/client.rs

```diff
@@ -112,19 +102,52 @@ impl<F: FieldElement> Client<F> {
         Ok((encrypted_share1, encrypted_share2))
     }
 
-    pub(crate) fn prove_with<G>(&mut self, init_function: G) -> Vec<F>
+    pub(crate) fn gen_proof(&mut self, input: &[F]) -> Vec<F> {
```

Comment:
I think this would be an opportune moment to add some doccomment explaining just what `input` is and what is returned from this function.